### PR TITLE
Code quality: Reduced duplicate code for the sidebar

### LINKED ISSUE #10102: Sidebar section creation has a lot of duplicated code
The method `GetOrCreateSection` in `src/Files.App/ViewModels/SidebarViewModel.cs` has a lot of duplicated code that could be refactored into simpler, cleaner functions.

Creating this issue to keep track of the refactoring.

### PR DESCRIPTION
**Pull request by QuaintMako**
**Resolved / Related Issues**

- Closes #10102 

**Validation**
How did you test these changes?
- [X] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

File: src/Files.App/ViewModels/SidebarViewModel.cs [hunk 1]
```diff
@@ -361,189 +373,103 @@ private async Task AddElementToSection(INavigationControlItem elem, LocationItem
...
-								SelectsOnInvoked = false,
-								Icon = new BitmapImage(new Uri("ms-appx:///Assets/FluentIcons/CloudDrive.png")),
-								ChildItems = new BulkConcurrentObservableCollection<INavigationControlItem>()
-							};
-							var index = sectionOrder.TakeWhile(x => x != sectionType).Select(x => SideBarItems.Any(item => item.Section == x) ? 1 : 0).Sum();
-							SideBarItems.Insert(Math.Min(index, SideBarItems.Count), section);
-						}
-						return section;
-					}
+			if(sectionType == SectionType.Home)
```
Comment (yaichenbaum) [inline src/Files.App/ViewModels/SidebarViewModel.cs:395]: Can we use a switch case?

**Comment by yaichenbaum** [comment]
Looks a lot cleaner 🎉

File: src/Files.App/ViewModels/SidebarViewModel.cs (hunk 1 above)
Comment (QuaintMako) [reply src/Files.App/ViewModels/SidebarViewModel.cs:395]: We can, but I used the if/else pattern so I could concatenated the type of the section and the condition for it to be displayed.  Can change it if you prefer.

Comment (yaichenbaum) [reply src/Files.App/ViewModels/SidebarViewModel.cs:395]: I think the if statements can be put inside the cases (break early if the condition isn't met).

File: src/Files.App/ViewModels/SidebarViewModel.cs [hunk 2]
```diff
@@ -361,189 +373,103 @@ private async Task AddElementToSection(INavigationControlItem elem, LocationItem
...
-								ChildItems = new BulkConcurrentObservableCollection<INavigationControlItem>()
-							};
-							var index = sectionOrder.TakeWhile(x => x != sectionType).Select(x => SideBarItems.Any(item => item.Section == x) ? 1 : 0).Sum();
-							SideBarItems.Insert(Math.Min(index, SideBarItems.Count), section);
-						}
-						return section;
-					}
-				default:
-					return null;
+				if(iconIdex != -1)
```
Comment (yaichenbaum) [inline src/Files.App/ViewModels/SidebarViewModel.cs:448]:
```suggestion
				if (iconIdex != -1)
```

File: src/Files.App/Constants.cs [hunk 3]
```diff
@@ -51,7 +51,25 @@ public static class ImageRes
             public const int Folder = 3;
         }
 
-        public static class Shell32
+		public static class FluentIconsPaths
```
Comment (yaichenbaum) [inline src/Files.App/Constants.cs:54]: Is spacing set to tabs?

File: src/Files.App/ViewModels/SidebarViewModel.cs (hunk 1 above)
Comment (QuaintMako) [reply src/Files.App/ViewModels/SidebarViewModel.cs:395]: Done.

File: src/Files.App/ViewModels/SidebarViewModel.cs (hunk 2 above)
Comment (QuaintMako) [reply src/Files.App/ViewModels/SidebarViewModel.cs:448]: Done.

File: src/Files.App/Constants.cs (hunk 3 above)
Comment (QuaintMako) [reply src/Files.App/Constants.cs:54]: It seems so. I'm wondering if the source code is not using tabs in some places and spaces in others.

Fixed.

**Comment by QuaintMako** [comment]
Closing that PR. I've messed up the commit history of that branch by fusing it with another PR. Reopening a new one.

**Pull request closed without merging**